tss-esapi/src/interface_types/data_handles.rs

```diff
@@ -0,0 +1,191 @@
+// Copyright 2023 Contributors to the Parsec project.
+// SPDX-License-Identifier: Apache-2.0
+/// This module contains native representations of the TPMI_DH types.
+use crate::{
+    handles::{
+        HmacSessionTpmHandle, NvIndexTpmHandle, PcrTpmHandle, PersistentTpmHandle,
+        PolicySessionTpmHandle, TpmHandle, TransientTpmHandle,
+    },
+    tss2_esys::{TPMI_DH_CONTEXT, TPMI_DH_SAVED},
+    Error, Result, WrapperErrorKind,
+};
+use std::convert::TryFrom;
+
+/// Enum representing the 'Object' data handles interface type.
+///
+/// # Details
+/// This corresponds to the TPMI_DH_OBJECT interface type.
+#[derive(Debug, Copy, Clone)]
+pub enum Object {
+    Transient(TransientTpmHandle),
+    Persistent(PersistentTpmHandle),
+}
+
+/// Enum representing the 'Parent' data handles interface type.
+///
+/// # Details
+/// This corresponds to the TPMI_DH_PARENT interface type.
+#[derive(Debug, Copy, Clone)]
+pub enum Parent {
+    Transient(TransientTpmHandle),
+    Persistent(PersistentTpmHandle),
+    Owner,
+    Platform,
+    Endorsement,
+}
+
+/// Enum representing the 'Persistent' data handles interface type.
+///
+/// # Details
+/// This corresponds to the TPMI_DH_PERSISTENT interface type.
+#[derive(Debug, Copy, Clone)]
+pub enum Persistent {
+    Persistent(PersistentTpmHandle),
```

Comment:
To be honest this one-variant enums look a bit weird... are they here because they need to wrap some TPM structure? Maybe a [newtype](https://doc.rust-lang.org/rust-by-example/generics/new_types.html) pattern would be more appropriate here :thinking:

Reply:
I do not think there us any special reason for this. I think I just did it that way for consistency. Every other interface type is an enum so I just did it the same way with those ones.